### PR TITLE
Call from native contract synchronously

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Review by shargon — DISMISSED** [review]
The code looks good, could @superboyiii test it with https://github.com/neo-project/neo-devpack-dotnet/pull/391 ?

**Comment by cloud8little** [comment]
It does not work. :(    Callingscripthash is still the contract hash.
2d8c1980809da636db510512a632a10

**Comment by shargon** [comment]
@cloud8little please try with the last patch

File: src/neo/SmartContract/ApplicationEngine.cs [hunk 1]
```diff
@@ -29,8 +29,6 @@ private enum ReturnTypeConvention : byte
 
         private class InvocationState
```
Comment (Qiao-Jin) [inline src/neo/SmartContract/ApplicationEngine.cs:30]: Maybe this class can be removed

File: src/neo/SmartContract/ApplicationEngine.cs [hunk 2]
```diff
@@ -84,20 +82,18 @@ protected override void OnFault(Exception e)
             base.OnFault(e);
         }
 
-        internal void CallFromNativeContract(Action onComplete, UInt160 hash, string method, params StackItem[] args)
+        internal void CallFromNativeContract(UInt160 callingScriptHash, UInt160 hash, string method, params StackItem[] args)
```
Comment (erikzhang) [inline src/neo/SmartContract/ApplicationEngine.cs:85]: I think we shouldn't modify the callingScriptHash. Because it may cause confusion.

**Comment by cloud8little** [comment]
> @cloud8little please try it again

The commit 4a5a8229 works. Mint with neo/gas multiple times successful.

beba8a833bcdd8200a23b386ae6730a

File: src/neo/SmartContract/ApplicationEngine.cs (hunk 2 above)
Comment (roman-khimov) [reply src/neo/SmartContract/ApplicationEngine.cs:85]: But we need for calling script hash to be correct for native invocations also. The problem now is that when native contracts call each other (natively), these calls are not reflected in any way on VM execution stack.

File: src/neo/SmartContract/ApplicationEngine.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.cs:30]: We can do it in another PR.

**Comment by erikzhang** [comment]
Merge?

**Review by shargon — APPROVED** [review]

**Pull request merged**